src/test/java/io/cdap/plugin/RunConfigTest.java

```diff
@@ -17,73 +17,104 @@
 package io.cdap.plugin;
 
 import io.cdap.cdap.api.data.schema.Schema;
-import io.cdap.cdap.etl.mock.common.MockPipelineConfigurer;
+import io.cdap.cdap.etl.api.validation.CauseAttributes;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.cdap.etl.api.validation.ValidationFailure;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
 /**
  * Test cases for RunConfig.
  */
-public class RunConfigTest extends TransformPluginsTestBase {
+public class RunConfigTest {
 
   private static final Schema inputSchema = Schema.recordOf("input-record",
                                                             Schema.Field.of("id", Schema.of(Schema.Type.STRING)),
                                                             Schema.Field.of("input", Schema.of(Schema.Type.STRING)));
+  private static final String MOCK_STAGE = "mockStage";
+  private static final RunConfig VALID_CONFIG = new RunConfig(
+    "java -jar /home/user/Example.jar",
+    "input",
+    "50 true",
+    "output",
+    "string",
+    null
+  );
+
+  @Test
+  public void testValidConfig() {
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    VALID_CONFIG.validate(failureCollector, inputSchema);
+    Assert.assertTrue(failureCollector.getValidationFailures().isEmpty());
+  }
 
   @Test
   public void testInvalidCommandToExecute() throws Exception {
-    Run.RunConfig config = new Run.RunConfig("java -jar", "input", "50 true", "output", "string", null);
-    MockPipelineConfigurer configurer = new MockPipelineConfigurer(inputSchema);
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    RunConfig config = RunConfig.builder(VALID_CONFIG)
+      .setCommandToExecute("java -jar")
+      .build();
+    List<List<String>> paramName = Arrays.asList(
+      Collections.singletonList(RunConfig.COMMAND_TO_EXECUTE),
+      Collections.singletonList(RunConfig.COMMAND_TO_EXECUTE));
+
     try {
-      new Run(config).configurePipeline(configurer);
-      Assert.fail();
-    } catch (IllegalArgumentException e) {
-      Assert.assertEquals("Error while accessing the binary. Please make sure that the 'Command to Execute' is in " +
-                            "the expected format. 'java -jar'", e.getMessage());
+      config.validate(failureCollector, inputSchema);
+    } catch (ValidationException e) {
+      assertValidationFailed(failureCollector, paramName);
```

Comment:
this test will not fail if validation passes but it should

Reply:
FIxed